From 6f8e1950411ec1d5a9777505deceb2da7dd0e963 Mon Sep 17 00:00:00 2001
From: Chee Yang Lee <chee.yang.lee@intel.com>
Date: Wed, 9 Nov 2022 15:04:56 +0800
Subject: [PATCH 1/2] configure.ac: remove AC_CHECK_FILE

AC_CHECK_FILE cannot check for file existence when cross compiling.

Upstream-Status: Inappropriate

Signed-off-by: Chee Yang Lee <chee.yang.lee@intel.com>
---
 configure.ac | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index fffb0bc..51d7790 100644
--- a/configure.ac
+++ b/configure.ac
@@ -257,9 +257,11 @@ AC_SUBST(enable_qat_auto_engine_init_on_fork)
 #Default library name is qatengine
 AC_SUBST([LIBQATNAME], "qatengine")
 
-AC_CHECK_FILE(${qat_hw_dir_prefix}/include/qat/icp_sal_versions.h,
-	      [with_icp_sal_versions_h=yes],
-	      [with_icp_sal_versions_h=no])
+if test -f "${qat_hw_dir_prefix}/icp_sal_versions.h"; then
+	with_icp_sal_versions_h=yes
+else
+	with_icp_sal_versions_h=no
+fi
 if test "x$with_icp_sal_versions_h" = "xyes"
 then
   if test `grep "define SAL_INFO2_DRIVER_SW_VERSION_TYPE \"in-tree\"" ${qat_hw_dir_prefix}/include/qat/icp_sal_versions.h | wc -l` = "1"
@@ -389,9 +391,11 @@ then
     then
       AC_SUBST([cflags_qat_sw], ["-DQAT_SW"])
       AC_SUBST([includes_qat_sw_crypto_mb], ["-I/usr/local/include"])
-      AC_CHECK_FILE(/usr/local/lib/libcrypto_mb.so,
-                   [AC_SUBST([QAT_SW_CRYPTO_MB_LIB], ["-Wl,-rpath,/usr/local/lib -L/usr/local/lib -lcrypto_mb"])],
-                   [AC_SUBST([QAT_SW_CRYPTO_MB_LIB], ["-lcrypto_mb"])])
+      if test -f "/usr/local/lib/libcrypto_mb.so"; then
+	      AC_SUBST([QAT_SW_CRYPTO_MB_LIB], [" -L/usr/local/lib -lcrypto_mb"])
+      else
+	      AC_SUBST([QAT_SW_CRYPTO_MB_LIB], ["-lcrypto_mb"])
+      fi
     fi
   fi
 
-- 
2.37.3

